src/backend/distributed/executor/multi_router_executor.c

```diff
@@ -958,13 +987,23 @@ ExecuteModifyTasks(List *taskList, bool expectResults, ParamListInfo paramListIn
 		return 0;
 	}
 
+	/*
+	 * In multi shard modification, we expect that all tasks operates on
+	 * same relation, so it is enough to acquire a lock on first relation's
```

Comment:
I think `on the first relation's partitions.` is not accurate? It should be `on the first task's anchor relation` or so?

`same relation` -> `the same relation`